packages/stream_chat_flutter/lib/src/message_input/attachment_preview/gallery_picker_screen.dart

```diff
@@ -0,0 +1,233 @@
+                        const Text(
+                          'Select from your phone gallery',
+                          style:
+                              TextStyle(color: UnikonColorTheme.dividerColor),
+                        ),
+                        const Spacer(),
+                        Text(
+                          '(${selectedIds.length}) Selected',
+                          style: const TextStyle(
+                              color: UnikonColorTheme.dividerColor),
+                        ),
+                      ],
+                    ),
+                  ),
+                  Expanded(
+                    child: GalleryPickerWidget(
+                      selectedMediaItems: selectedIds,
+                      onMediaItemSelected: (AssetEntity media) async {
+                        if (selectedIds.contains(media.id)) {
+                          return await attachmentController
+                              .removeAssetAttachment(media);
+                        }
+                        await attachmentController.addAssetAttachment(media);
+                      },
+                    ),
+                  ),
+                ],
+              ),
+              if (attachmentController.value.isNotEmpty)
+                Positioned(
+                  bottom: 10,
+                  right: 10,
+                  child: GestureDetector(
+                    onTap: () {
+                      Navigator.pushReplacement(
+                        context,
+                        MaterialPageRoute(
+                          builder: (context) => AttachmentPreviewScreen(
+                            attachmentController: attachmentController,
+                            effectiveController: widget.effectiveController,
+                          ),
+                        ),
+                      );
+                    },
+                    child: Container(
+                      padding: const EdgeInsets.symmetric(
+                          horizontal: 36, vertical: 13),
+                      decoration: BoxDecoration(
+                          color: UnikonColorTheme.primaryColor,
+                          borderRadius: BorderRadius.circular(100)),
+                      child: const Text(
+                        'Next',
+                        style: TextStyle(
+                          color: UnikonColorTheme.messageSentIndicatorColor,
+                        ),
+                      ),
+                    ),
+                  ),
+                )
+            ],
+          ),
+        );
+      },
+    );
+  }
+}
+
+class BuildMediaAttachment extends StatelessWidget {
+  const BuildMediaAttachment({
+    super.key,
+    required this.effectiveController,
+  });
+  final StreamMessageInputController effectiveController;
+
+  @override
+  Widget build(BuildContext context) {
+    StreamAttachmentPickerController attachmentController =
+        StreamAttachmentPickerController();
```

Comment:
Can't we move this initialisation outside of build method since it will try to create new obj everytime it rebuilds.

Reply:
It's a stateless widget, so not required here.